### ISSUE TITLE
With label and annotation filters before a Watch, a pod is getting watched that does not have filters

### ISSUE THREAD

**Issue by cmwylie19** [defenseunicorns/pepr#355]
### Environment
Device and OS:
App version:
Kubernetes distro being used:
Other:

### Steps to reproduce
Create a Pepr module with the following capability:

```typescript
import { KubeConfig, Exec } from "@kubernetes/client-node";
import { Capability, a, Log, R } from "pepr";

export const IstioJobTermination = new Capability({
  name: "istio-job-termination",
  description: "Ensure Istio sidecars are terminated after job completion",
});

// Use the 'When' function to create a new action
const { When } = IstioJobTermination;

When(a.Pod)
  .IsCreatedOrUpdated() // IsCreated doesn't trigger enough :thinking:
  .WithLabel("batch.kubernetes.io/job-name")
  // .WithLabel("service.istio.io/canonical-name")
  .WithAnnotation("sidecar.istio.io/status")
  .Watch(async pod => {
    Log.info("watch triggered for " + pod.metadata.name)
    if (pod.status.phase == "Running") {
      const podReadyForTermination = R.all(containerStatus => {
        return (
          containerStatus.state.terminated?.exitCode == 0 ||
          containerStatus.name == "istio-proxy"
        );
      })(pod.status.containerStatuses);

      if (podReadyForTermination) {
        Log.info("Attempting to terminate sidecar for " + pod.metadata.name);
        try {
          const kc = new KubeConfig();
          kc.loadFromDefault();
          const exec = new Exec(kc);

          await exec.exec(
            pod.metadata.namespace,
            pod.metadata.name,
            "istio-proxy",
            ["pilot-agent", "request", "POST", "/quitquitquit"],
            null, // Could capture exec stdout here
            null, // Could capture exec stderr here
            process.stdin,
            true,
          );
        } catch (error) {
          // This is buggy, too many watch triggers, exec will fail once pod terminates
          Log.error(error, "Failed to terminate the pod sidecar");
        }
      }
    }
  });
  ```

### Expected result
Only pods meeting both filters to be watched.

### Actual Result
He is watching logs and seeing the "watched triggered for ..." on a pod that does not meet have the annotation in the capability binding.

Reported by @mjnagel 

### Additional Context
Add any other context or screenshots about the technical debt here.

**Comment by jeff-mccoy** [comment]
We need to add something for this actually as annotations aren't a valid field selector for watches, but it's not a super easy thing to prevent in typescript without a major overall of the types.  Tbh I thought I pushed a fix for this a few weeks ago but must be losing my mind. The filters have been updated, but think because we use interface vs type, TS isn't yelling in Pepr:

https://github.com/defenseunicorns/kubernetes-fluent-client/blob/main/src/fluent/types.ts#L23

Tl;dr; sadly annotation filtering is not actually a thing for the watch as the K8s API only supports label selectors & field selectors--but annotations aren't valid field selectors.

**Comment by cmwylie19** [comment]
Doing some backlog cleanup and I tested this and it seems to not be a problem anymore. There was a development to solve this on March 8: https://github.com/defenseunicorns/pepr/commit/3d8db0e811b93ffb14d68d1e2fe3366a1d4e853f#diff-e4da12f8a4bcd9ff17c90ed0ed4b7c1206712190cd9a3dfeb4d7656f8f8ea897.

All my testing is indicating that this is fixed now, feel free to open it up if you see this.
